cmd/boulder/main_test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/test"
+)
+
+// TestConfigValidation checks that each of the components which register a
+// validation tagged Config struct at init time can be used to successfully
+// validate their corresponding test configuration files.
+func TestConfigValidation(t *testing.T) {
```

Comment:
It feels a little bit odd for this to be a unit test rather than an integration test. I don't think there are any other unit tests thus far which reference `//test/config[-next]/[service].json`. And at the end of the day, you're not just testing "does each part of the config validation framework do the right thing", you're testing "does config validation work end-to-end on real-ish data", which feels like an integration test.

Reply:
Yeah, I honestly didn't feel great about it either. I actually started with an integration test. Though I quickly realized that I wouldn't be able to accomplish it in integration by importing boulder code. You need to import each of the cmd packages to trigger their `init` functions, and I don't believe that's possible. I considered a couple of other options but this approach gave me the best mix of convenience and safety.

Reply:
Roger, yep, this is a convincing reason to leave it here. I think it'll make more sense here once cmd/boulder/main.go is handling the `-validate` flag, too.